File: polocloud-suite/src/main/java/dev/httpmarco/polocloud/suite/commands/type/PlatformArgument.java
package dev.httpmarco.polocloud.suite.commands.type;

import dev.httpmarco.polocloud.suite.PolocloudSuite;
import dev.httpmarco.polocloud.suite.commands.CommandArgument;
import dev.httpmarco.polocloud.suite.commands.CommandContext;
import dev.httpmarco.polocloud.suite.platforms.Platform;

import java.util.List;

public final class PlatformArgument extends CommandArgument<Platform> {

    public PlatformArgument(String key) {
        super(key);
    }

    @Override
    public List<String> defaultArgs(CommandContext context) {
        return PolocloudSuite.instance().platformProvider().platforms().stream().map(Platform::name).toList();
    }

    @Override
    public Platform buildResult(String input) {
        return PolocloudSuite.instance().platformProvider().findPlatform(input);
    }
}

File: agent/src/main/kotlin/dev/httpmarco/polocloud/agent/runtime/local/terminal/arguments/type/PlatformArgument.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.type

import dev.httpmarco.polocloud.agent.i18n
import dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.TerminalArgument
import dev.httpmarco.polocloud.agent.runtime.local.terminal.arguments.InputContext
import dev.httpmarco.polocloud.platforms.Platform
import dev.httpmarco.polocloud.platforms.PlatformPool

class PlatformArgument(key: String = "platform") : TerminalArgument<Platform>(key) {

    override fun buildResult(input: String, context: InputContext): Platform {
        return PlatformPool.find(input)!!
    }

    override fun wrongReason(rawInput: String): String {
        return i18n.get("agent.terminal.setup.argument.platform.wrong")
    }

    override fun defaultArgs(context: InputContext): MutableList<String> {
        return PlatformPool.platforms().stream().map { it.name }.toList()
    }

    override fun predication(rawInput: String): Boolean {
        return PlatformPool.find(rawInput) != null
    }
}